lib/membrane_rtc_engine/endpoints/hls_endpoint.ex

```diff
@@ -224,90 +212,166 @@ if Enum.all?(
       track_children =
         [
           :opus_decoder,
+          :aac_encoder,
+          :aac_parser,
           :keyframe_requester,
           :video_parser,
+          :video_parser_out,
           :decoder,
+          :encoder,
+          :resolution_scaler,
           :framerate_converter
         ]
         |> Enum.map(&{&1, track_id})
         |> Enum.filter(&Map.has_key?(ctx.children, &1))
 
-      {_removed_track, tracks} = Map.pop!(state.tracks, track_id)
+      {removed_track, tracks} = Map.pop!(state.tracks, track_id)
       state = %{state | tracks: tracks}
 
-      children_to_remove = track_children ++ if tracks == %{}, do: get_common_children(), else: []
+      children_to_remove =
+        if is_nil(state.mixer_config) do
+          sink_bin_used? =
+            Enum.any?(tracks, fn {_id, track} ->
+              track.stream_id == removed_track.stream_id
+            end)
+
+          if sink_bin_used?,
+            do: track_children,
+            else: [{:hls_sink_bin, removed_track.stream_id} | track_children]
+        else
+          track_children ++ if tracks == %{}, do: get_common_children(), else: []
+        end
```

Comment:
What do you think about this change?
```suggestion
      sink_bin_used? =
            Enum.any?(tracks, fn {_id, track} ->
              track.stream_id == removed_track.stream_id
            end)
      children_to_remove = cond do 
        is_nil(state.mixer_config) and sink_bin_used? ->  [{:hls_sink_bin, removed_track.stream_id}]
        not is_nil(state.mixer_config) and tracks == %{} -> get_common_children()
        true -> []
        end
        children_to_remove = track_children ++ children_to_remove
```